easybuild/easyconfigs/c/CGAL/CGAL-4.11-intel-2017b-Python-3.6.3.eb

```diff
@@ -0,0 +1,37 @@
+name = 'CGAL'
+version = '4.11'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://www.cgal.org/'
+description = """The goal of the CGAL Open Source Project is to provide easy access to efficient
+ and reliable geometric algorithms in the form of a C++ library."""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+toolchainopts = {'strict': True}
+
+source_urls = ['https://github.com/CGAL/cgal/releases/download/releases%2FCGAL-%(version)s']
+sources = [SOURCE_TAR_XZ]
+checksums = ['27a7762e5430f5392a1fe12a3a4abdfe667605c40224de1c6599f49d66cfbdd2']
+
+dependencies = [
+    ('zlib', '1.2.11'),
+    ('Python', '3.6.3'),
+    ('Boost', '1.65.1', versionsuffix),
+    ('MPFR', '3.1.5'),
+    ('GMP', '6.1.2'),
+    ('libGLU', '9.0.0'),
+    ('Qt5', '5.8.0'),
```

Comment:
@Helios07 Existing `CGAL` 4.11 easyconfigs use `Qt` 4.x rather than `Qt5`, can you explain the jump to `Qt5`, do you have a specific reason for it?

In addition, there's a `Qt5` 5.10.0 (and 5.9.x) already, why not go with the most recent `Qt5`?

Reply:
@boegel Qt 5.9 was not building for my setup yet, so I kept using 5.8. I am not sure about the differences between Qt 4 and 5 so I tried to use Qt 5. Would you suggest to stick with version 4 here?

Reply:
It probably makes sense to go forward with `Qt5`, but then using a more recent version would make sense.

In #5344, the `CGAL` easyconfig uses `Qt5` 5.9.2 (but there's a 5.9.3 already too).

Reply:
Qt 5.9 won't build on my CentOS 7 system with EasyBuild. I am searching for a solution.

Reply:
@Helios07 Does https://github.com/easybuilders/easybuild-easyblocks/pull/1328 help?

Reply:
In fact, it does :)

Reply:
@Helios07 Any updates?

Reply:
@boegel I still cannot build Qt 5.9.3 with the Intel toolchain and the error messages are not helpful. It worked with foss, but I cannot find, what I am doing wrong.

Reply:
Are you using EasyBuild v3.5.1? Can you share the build log with the errors somewhere, e.g. via gist.github.com?

Reply:
Yes, I use EasyBuild 3.5.1. You can fine my build log here: 
https://uni-muenster.sciebo.de/s/jNcjLmsNizqJw9g

The output of the build was:
[h_zimm01@palma003 ~]$ eb Qt5-5.9.3-intel-2017b.eb
== temporary log file in case of crash /tmp/eb-3lw6m3/easybuild-d2LMFx.log
== processing EasyBuild easyconfig /home/h/h_zimm01/Qt5-5.9.3-intel-2017b.eb
== building and installing MPI/intel/2017.4.196-GCC-6.4.0-2.28/impi/2017.3.196/Qt5/5.9.3...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== FAILED: Installation ended unsuccessfully (build directory: /tmp/easybuild/build/Qt5/5.9.3/intel-2017b): build failed (first 300 chars): cmd " LD_LIBRARY_PATH=/tmp/easybuild/build/Qt5/5.9.3/intel-2017b/qt-everywhere-opensource-src-5.9.3/qtbase/lib:/tmp/easybuild/build/Qt5/5.9.3/intel-2017b/qt-everywhere-opensource-src-5.9.3/qtdeclarative/lib:$LD_LIBRARY_PATH  make -j 32 " exited with exit code 2 and output:
cd qtbase/ && ( test -e Ma
== Results of the build can be found in the log file(s) /tmp/eb-3lw6m3/easybuild-Qt5-5.9.3-20180119.142624.YhlqZ.log
ERROR: Build of /home/h/h_zimm01/Qt5-5.9.3-intel-2017b.eb failed (err: 'build failed (first 300 chars): cmd " LD_LIBRARY_PATH=/tmp/easybuild/build/Qt5/5.9.3/intel-2017b/qt-everywhere-opensource-src-5.9.3/qtbase/lib:/tmp/easybuild/build/Qt5/5.9.3/intel-2017b/qt-everywhere-opensource-src-5.9.3/qtdeclarative/lib:$LD_LIBRARY_PATH  make -j 32 " exited with exit code 2 and output:\ncd qtbase/ && ( test -e Ma')

Reply:
Eek, that's a nasty error...

```
In file included from deps/geojsonvt/6.3.0/include/mapbox/geojsonvt/simplify.hpp(3),
                 from deps/geojsonvt/6.3.0/include/mapbox/geojsonvt/convert.hpp(3),
                 from deps/geojsonvt/6.3.0/include/mapbox/geojsonvt.hpp(3),
                 from src/mbgl/annotation/shape_annotation_impl.hpp(4),
                 from src/mbgl/annotation/fill_annotation_impl.hpp(3),
                 from src/mbgl/annotation/fill_annotation_impl.cpp(1):
deps/geojsonvt/6.3.0/include/mapbox/geojsonvt/types.hpp(63): error: cannot determine the exception specification of the default constructor due to a circular dependency
      using container_type::container_type;
```

No idea what's going on here, but it looks like this may be a compiler bug.

This, on top of the problem with using `Qt5` as indirect dependency we're running into in #5344 make me reconsider the switch to `Qt5`, I think we should stick with the latest `Qt` 4.x instead?

I've just opened a PR for Qt 4.8.7 with `intel/2017b` at #5682, can you try changing the `Qt` in this `CGAL` easyconfig to `4.8.7`, and using that instead?

Reply:
No problem. It also builds with your Qt 4.8.7 package.